specs/carousel/carousel.md

```diff
@@ -0,0 +1,258 @@
+
+*CSS custom properties:*
+- ???
+
+*Slots:*
+- `default` - slot for slides content // TODO: IS this the right way?
+- `play-toggle` - the control to toggle if the slides rotate or not
+- `next-flipper` - the control used to rotate to the next slide
+- `previous-flipper` - the control used to rotate to the previous slide
+- `tablist` - the control to select a certain slide
+
+*Events*
+- `on-pause` - callback fired when the rotation is stopped, such as when keyboard focus or hover is on any element of the carousel, or the *play-toggle* is set to off.
+- `on-active-id-update` - callback fired when the active id is changed
+
+### Anatomy and Appearance
+**Carousel Structure:**
+
+Note: the Carousel class will have a method that checks the attribute names and grabs the items.
+
+```html
+<template>
+    <div
+        class="carousel"
+        aria-roledescription="carousel"
+        tabindex="-1"
+    >
+    <slot name="play-toggle"> SEE DEFAULT </slot>
+    <slot name="next-flipper"> SEE DEFAULT </slot>
+    <div
+        class="carousel-items"
+        aria-live="off"
+    >
+        <!-- Solution 1 -->
+        <!-- Map all items and use css class's to display correct item based on id. -->
+        <!-- May be a less performant way as the items will be mapped each time. -->
+        ${ x => x.items.map(item, index => {
+            return `
+            <div
+                class="carousel-item ${x => x.getActiveIndex() === index ? 'active' : ''}"
+                role="group"
+                aria-roledescription="slide"
+                aria-label="${x => x.getActiveIndex()} of ${x => x.getItemCount()}"
+            >
+                ${item}
+            </div>`
+            }).join('')
+        }
+        <!-- Solution 2 -->
+        <!-- Have carousel class method that query selects this div and inserts content for the correct item based on id/index from the id's given as an array. -->
+        <div
+            class="carousel-item"
+            role="group"
+            aria-roledescription="slide"
+            aria-label="${x => x.getActiveIndex()} of ${x => x.getItemCount()}"
+        >
+            ${x => x.getActiveItem()}
+        </div>
+    </div>
+    <slot name="previous-flipper"> SEE DEFAULT </slot>
+    <slot name="tab-list">SEE DEFAULT </slot>
+    </div>
+</template>
+```
+
+**Default Flipper**
+```html
+<button
+    @click="${x => x.next()}"
+    class="next-flipper"
+>
+    <svg class="flipper-svg">
+        <path d="example path..."></path>
+    </svg>
+</button>
+```
+
+**Default Play Toggle**
+```html
+<button
+    slot="play-toggle"
+    @click="${x => x.togglePlay()}"
+    class="previous-flipper"
+>
+    <svg class="play-svg">
+        <path d="example path..."></path>
+    </svg>
+</button>
+```
+
+**Default Tab List**
+```html
+<div role="tablist" class="tablist" aria-label="A carousel of items">
```

Comment:
We will want to use whatever we decide for `fast-tabs` here. I just started looking into that spec. Maybe we should sync up on this.

Reply:
I think a sync might be helpful here, but I'm not 100% sure we need to map to `tabs`. There is a working draft for carousel in the W3C as a control: https://w3c.github.io/aria-practices/#carousel

I do have some thoughts on the above spec though as it seems to include several interaction models that conflict and could make implementing difficult or confusing for end users.

Reply:
Yes, I saw that the old one uses tabs, but feel like it may not be necessary. I'll address some of these other issues and then sync with you both on the above.

Reply:
I have added a simple tablist example in the **implementation** of the spec.